osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverter.java

```diff
@@ -38,28 +44,89 @@
-        final String[] values = source.split(";");
-        if (values.length == SIMPLE_END_INDEX) {
-            stringArrayToAnalogList = new SimpleStringToAnalogList();
-        } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
-            stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
-        } else {
-            LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+        LOGGER.info("Source string: {}", source);
+
+        try {
+            final Payload payload = this.objectMapper.readValue(source, Payload[].class)[0];
+
+            final String[] values = (payload.gisnr + ", " + String.join(", ", payload.data)).split(", ");
+            LOGGER.info("Values length: {} and values: {}", values.length, Arrays.toString(values));
+
+            if (values.length == SIMPLE_END_INDEX) {
+                stringArrayToAnalogList = new SimpleStringToAnalogList();
+            } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
+                stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
+            } else {
+                LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+                return null;
+            }
+
+            final String eanCode = values[0];
+            final PowerSystemResource powerSystemResource = new PowerSystemResource(eanCode,
+                    UUID.randomUUID().toString(), new ArrayList<Name>());
+            final long createdDateTime = System.currentTimeMillis();
+            return new GridMeasurementPublishedEvent(createdDateTime, eanCode, UUID.randomUUID().toString(),
+                    "GridMeasurementPublishedEvent", new ArrayList<Name>(), powerSystemResource,
+                    stringArrayToAnalogList.convertToAnalogList(values));
+        } catch (final JsonMappingException e) {
+            LOGGER.error("Caught an error mapping a JSON string to Payload. {}", source);
+            e.printStackTrace();
+            return null;
+        } catch (final JsonProcessingException e) {
+            LOGGER.error("Caught an error processing a JSON string to Payload. {}", source);
+            e.printStackTrace();
             return null;
         }
+    }
 
-        final String eanCode = values[0];
-        final PowerSystemResource powerSystemResource = new PowerSystemResource(eanCode, UUID.randomUUID().toString(),
-                new ArrayList<Name>());
-        final long createdDateTime = System.currentTimeMillis();
-        return new GridMeasurementPublishedEvent(createdDateTime, eanCode, UUID.randomUUID().toString(),
-                "GridMeasurementPublishedEvent", new ArrayList<Name>(), powerSystemResource,
-                stringArrayToAnalogList.convertToAnalogList(values));
+    private static int ordinalIndexOf(final String str, final String substr, int n) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "ordinalIndexOf" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)